pyannote/database/registry.py

```diff
@@ -0,0 +1,618 @@
+    
+    """
+
+    def __init__(self) -> None:
+
+        # Mapping of database.yml paths to their config in a dictionary
+        # Example after loading both database.yml:
+        #   {"/path/to/first/database.yml": {
+        #           "Databases":{
+        #               "DatabaseA": ["relative/path/A/trn/{uri}.wav", "relative/path/A/dev/{uri}.wav", relative/path/A/tst/{uri}.wav]
+        #               "DatabaseB": "/absolute/path/B/{uri}.wav" 
+        #           },
+        #           "Protocols":{
+        #               "DatabaseA":{
+        #                   "SpeakerDiarization": {
+        #                       "ProtocolA": {
+        #                           "train": {"uri": "relative/path/A/trn.lst"},
+        #                           "development": {"uri": "relative/path/A/dev.lst"},
+        #                           "test": {"uri"; "relative/path/A/tst.lst"}
+        #                       }
+        #                   }
+        #               },
+        #               "DatabaseB":{"SpeakerDiarization":{"Protocol": {...}}},
+        #               "X":{"SpeakerDiarization":{"A_and_B":{...}}}
+        #           }
+        #       },
+        #    "/path/to/second/database.yml": {
+        #           "Databases":{
+        #               "DatabaseC": /absolute/path/C/{uri}.wav
+        #               "DatabaseB": "/absolute/path/B/{uri}.wav" 
+        #           },
+        #           "Protocols":{
+        #               "DatabaseB":{"SpeakerDiarization": {"Protocol": {...}}},
+        #               "DatabaseC":{...}
+        #           }
+        #       }
+        #   }
+        self.configs: Dict[Path, Dict] = dict()
+
+
+        # Content of the "Database" root item (= where to find file content)
+        # Example after loading both database.yml:
+        #   {
+        #   "DatabaseA": [
+        #       "/path/to/first/relative/path/A/trn/{uri}.wav",
+        #       "/path/to/first/relative/path/A/dev/{uri}.wav",
+        #       /path/to/first/relative/path/A/tst/{uri}.wav
+        #       ],
+        #   "DatabaseB": ["/absolute/path/B/{uri}.wav"],
+        #   "DatabaseC": ["/absolute/path/C/{uri}.wav"]
+        #   }
+        self.sources: Dict[Text, List[Text]] = dict()
+
+        # Mapping of database names to a type that inherits from Database
+        # Example after loading both database.yml:
+        #   {"DatabaseA": pyannote.database.registry.DatabaseA,
+        #   "DatabaseB": pyannote.database.registry.DatabaseB,
+        #   "DatabaseC": pyannote.database.registry.DatabaseC,
+        #   "X": pyannote.database.registry.X}
+        self.databases: Dict[Text, Type] = dict()
+
+        # Mapping of tasks name to the set of databases that support this task
+        # Example after loading both database.yml:
+        #   {"SpeakerDiarization", {"DatabaseA", "DatabaseB", "DatabaseC", "X"}}
+        self.tasks: Dict[Text, Set[Text]] = dict()
```

Comment:
I don't think I have ever found a use case for this task-based filtering thing.
I suggest we remove this completely -- this will make code cleaner anyway.

Reply:
The only uses I've found are :
- `get_databases(task=None)` where the task filter databases that do not "implement" this task
- `get_tasks()`, which is used in the client with `pyannote-database task [DATABASE]` when DATABASE is not passed (it then lists all tasks in the database.yml)

Should these methods / behaviour stay ? (by reimplementing registry.get_tasks() so that it computes on the fly what was previously cached in `registry.tasks`)